src/storage/patch/RepresentationPatchHandler.ts

```diff
@@ -37,10 +46,65 @@ export class RepresentationPatchHandler extends PatchHandler {
       }
       this.logger.debug(`Patching new resource ${identifier.path}`);
     }
-
+    // Converter of RDF text to N3 store
+    const converter = new RdfToQuadConverter();
+    let clonedRepresentation;
+    if (representation) {
+      clonedRepresentation = await cloneRepresentation(representation);
+    }
     // Patch it
     const patched = await this.patcher.handleSafe({ patch, identifier, representation });
 
+    // Solid, §5.3: "Servers MUST NOT allow HTTP POST, PUT and PATCH to update a container’s resource metadata
+    // statements; if the server receives such a request, it MUST respond with a 409 status code.
+    // https://solid.github.io/specification/protocol#contained-resource-metadata-statements
+    if (isContainerIdentifier(identifier)) {
+      throw new ConflictHttpError('Not allowed to execute PATCH request on containers.');
+    }
+
+    if (identifier.path.endsWith('.meta')) {
+      // Convert RDF representation to N3
```

Comment:
In this class we do not know if the resource is an RDF representation or not. Currently we only support PATCH on RDF representations because we only support SPARQL UPDATE and N3 Patch, but there is nothing that says that we can't support PATCH on non-RDF resources in the future with a different PATCH type.

Besides that we don't really want to clone representations if we can avoid it. It is a very expensive operation. It also feels like it is not needed, there is a point in the PATCH process where we have quads, so we should do these checks somewhere during that point. We might want something like an `RdfPatcher` that sits inbetween the N3Patcher/SparqlUpdatePatcher and its parent patcher and handles everything related to this, as those patchers take as input a stream of quads and return one again.